tools/meteor-services/deploy.js

```diff
@@ -321,6 +334,122 @@ function canonicalizeSite(site) {
   return parsed.hostname;
 };
 
+// Executes the poll to check for deployment success and outputs proper messages
+// to user about the status of their app during the polling process
+function executePollForDeploymentSuccess(versionId, deployPollTimeout, result) {
+  // Create a default polling configuration for polling for deploy / build
+  // In the future, we may change this to be user-configurable or smart
+  // The user can only currently configure the polling timeout via a flag
+  let pollingState = new PollingState(deployPollTimeout);
+  
+  var deploymentPollResult = buildmessage.enterJob({
+    title: "Waiting for deploy to succeed..."},
+    function() {
+      return pollForDeploy(pollingState,
+        versionId);
+    });
+
+  if (deploymentPollResult && deploymentPollResult.isActive) {
```

Comment:
So I'm not sure how I feel about using "is active" (as opposed to "succeeded") as the success condition. If you haven't configured your CI to only run one commit at a time on your deploy-to-production branch (which is arguably a mistake), then if you push a commit and then quickly push another commit, it probably shouldn't make your CI go red just because Galaxy decided that the later one should override the former one after the former one successfully builds?

Though I dunno, maybe it's just plain broken to set your CI up in a way where it doesn't serialize your deploy branch.

Reply:
Thinking about this in another vein, if this isn't the final step in CI, but it then maybe starts up an integration test suite that interacts with their newly deployed app, they'd want to make sure that the newly deployed app is the one that they're talking to.